### PR TITLE
Fix for packages sharing the same dependencies

### PR DESCRIPTION
**Pull request by jacobshirley**
Hi there, I was running into issues where, if multiple packages depend on the same packages, only the first one in the list will receive them. By removing the performance code in the resolver, it appears to be fixed. I've added a test, which verifies the nginx has a module that is required by both 'nginx-full' and 'nginx-core', when only 'nginx-core' goes into the image.

Please feel free to close this PR and do a better fix if there is one.

### CODE REVIEW

**Comment by loosebazooka** [comment]
@jacobshirley we require a signed CLA before proceeding

**Review by thesayyn — APPROVED** [review]
Can you move the new e2e test into examples/shared_dependencies?

**Comment by jacobshirley** [comment]
Sure, that's done now

File: examples/shared_dependencies/MODULE.bazel [hunk 1]
```diff
@@ -0,0 +1,25 @@
+bazel_dep(name = "rules_distroless", version = "0.0.0", dev_dependency = True)
```
Comment (thesayyn) [inline examples/shared_dependencies/MODULE.bazel:1]: You don't need a MODULE.bazel here, you can just use existing MODULE at root

File: examples/shared_dependencies/MODULE.bazel.lock [hunk 2]
```diff
@@ -0,0 +1,3543 @@
+{
```
Comment (thesayyn) [inline examples/shared_dependencies/MODULE.bazel.lock:1]: remove the lock please

File: examples/shared_dependencies/WORKSPACE.bazel [hunk 3]
```diff
@@ -0,0 +1,46 @@
+workspace(name = "shared_dependencies")
```
Comment (thesayyn) [inline examples/shared_dependencies/WORKSPACE.bazel:1]: Also remove the workspace here.

**Comment by jacobshirley** [comment]
Apologies for the delay on this. Hopefully that's looking ok now

**Review by thesayyn — APPROVED** [review]

**Comment by thesayyn** [comment]
Thank you!

**Pull request merged**